### ISSUE TITLE
Not Authenticating

### ISSUE THREAD

**Issue by jamesgagliardo** [RittmanMead/insights#7]
Hello,  I followed the install instructions and was able to navigate to the login page.  However, I'm getting authentication error for any type of user when logging in.  I've verified the OBIEE web services are working properly outside of Insights.  Any help would be appreciated.

**Comment by mineshpatel1** [comment]
This is the second time recently that I have seen this behaviour, although that time we couldn't verify the web services outside of Insights, so might need a bit more information. So the application reports "Authentication Error" whenever you attempt to login?

When you verified the web services outside of Insights, did you use the same login credentials?

Is there any peculiar OBIEE security set up on the system?

**Comment by jamesgagliardo** [comment]
Thanks for getting back to me.  Yes,  I’m getting ‘Authentication Error’ when I attempt to login.  I’ll dive in a bit more tomorrow when I’m in the office to see what errors it may be logging in Enterprise Manager.

Jim Gagliardo
Business Intelligence Developer
jgagliardo@homeadvisor.com<mailto:jgagliardo@homeadvisor.com>
303-963-8961
[HA logo]

**Comment by jamesgagliardo** [comment]
I solved this issue (sort of) by giving explicit rights to the BI Consumer group for each SOAP compenents under privledges.

**Comment by vrpkoduri** [comment]
We have AD/OAM Authentication enabled - We are getting Authentication Error. Any help on how do we solve this issue?
We are able to login to OBIEE with out issues. When we set Httpsonlycookies tag in instanceconfig.xml then we are not even able to login to OBIEE. 

Please let us know how we can resolve the issue

**Comment by mineshpatel1** [comment]
Integration with AD should work fine. Please test you can access the web service directly (e.g. using SOAPUI or Fiddler) and see if there is a similar error.

**Comment by jamesgagliardo** [comment]
I found another issue with this great tool that I hope can be resolved.  Any objects in subfolders of the subject area do not show up in the data source.  Is there anything I need to do to enable that?

**Comment by mineshpatel1** [comment]
That is peculiar. By any chance, are you using the old syntax for declaring child objects in the presentation layer, or are you using the Child Tables feature? Only the latter is supported.

Also, please start a new issue for different problems, it is extremely helpful.

**Comment by jamesgagliardo** [comment]
Thanks.  Let me check on this.  I’ll start a new issue for this.

**Comment by sunnyobiee** [comment]
Is this authentication issue resolved ? If yes please let us know what steps need to be taken . Some time I was able to login but subject area pane keeps spinning .

**Comment by jamesgagliardo** [comment]
Yes.  I was able to authenticate eventually.   I had to add the user’s group to several privilege components under SOAP. Not sure exactly which one finally worked.  We do,  however, have an outstanding issue with Insights where subfolders within the subject area do not show up.

Jim Gagliardo
Business Intelligence Developer
jgagliardo@homeadvisor.com<mailto:jgagliardo@homeadvisor.com>
303-963-8961
[HA logo]

**Comment by sunnyobiee** [comment]
Can you share a screenshot of the steps you did. I'm logging in as weblogic user and i added this user to all the SOAP Components
![capture](https://user-images.githubusercontent.com/33485562/32563123-f89d83ae-c475-11e7-9af3-acbbcfb7c94c.JPG)
 that i can see under privileges tab .Still i'm not able to login .

**Comment by serdarguler** [comment]
The problem is about obiee 12c is no longer supporting getgroups service. I could manage to handle it with a little touch in the obiee.js "obiee.getAppRoles" javascript function.

**Comment by spkoes** [comment]
Same problem in 12.2.1.3. @serdarguler , which change have you did in obiee.js?

**Comment by ddwolf715** [comment]
Same problem in 12.2.1.x. @serdarguler , which change have you did in obiee.js?

**Comment by ddwolf715** [comment]
> The problem is about obiee 12c is no longer supporting getgroups service. I could manage to handle it with a little touch in the obiee.js "obiee.getAppRoles" javascript function.
Same problem in 12.2.1.x. @serdarguler , which change have you did in obiee.js?

**Comment by ddwolf715** [comment]
> The problem is about obiee 12c is no longer supporting getgroups service. I could manage to handle it with a little touch in the obiee.js "obiee.getAppRoles" javascript function.



Same problem in 12.2.1.x. @serdarguler , which change have you did in obiee.js?

**Comment by serdarguler** [comment]
@ddwolf715 @spkoes 
```
obiee.getAppRoles = function (successFunc, errFunc) {
        var soapMessage = obieeSOAPHeader();
        soapMessage += '<soapenv:Body><' + wsdl + ':getGroups><' + wsdl + ':member><' + wsdl + ':name>';
        soapMessage += sessionStorage.obieeUser + '</' + wsdl + ':name></' + wsdl + ':member>';
        soapMessage += '<' + wsdl + ':expandGroups>False</' + wsdl + ':expandGroups>';
        soapMessage += '<' + wsdl + ':sessionID>' + sessionStorage.obieeSessionId + '</' + wsdl + ':sessionID>';
        soapMessage += '</' + wsdl + ':getGroups></soapenv:Body></soapenv:Envelope>';
        wsCall('securityService', soapMessage, function (response) {
            var appRoles = [];
            var results = response.Body.getGroupsResult || { account: [] };
            if (results.account) {
                results.account.forEach(function (r) {
                    if (r.accountType == '4')
                        appRoles.push({ id: r.name, name: r.displayName });
                });
            }
            appRoles.push({ id: "BIServiceAdministrator", name: "BIServiceAdministrator" });
            appRoles.push({ id: "BIContentAuthor", name: "BIContentAuthor" });
            appRoles.push({ id: "BIConsumer", name: "BIConsumer" });
            appRoles.push({ id: "AuthenticatedUser", name: "AuthenticatedUser" });
            successFunc(appRoles);
        }, function (err) {
            var appRoles = [];
            appRoles.push({ id: "BIServiceAdministrator", name: "BIServiceAdministrator" });
            appRoles.push({ id: "BIContentAuthor", name: "BIContentAuthor" });
            appRoles.push({ id: "BIConsumer", name: "BIConsumer" });
            appRoles.push({ id: "AuthenticatedUser", name: "AuthenticatedUser" });
            successFunc(appRoles);
        });
    }
```

**Comment by ddwolf715** [comment]
> @ddwolf715 @spkoes
> 
> ```
> obiee.getAppRoles = function (successFunc, errFunc) {
>         var soapMessage = obieeSOAPHeader();
>         soapMessage += '<soapenv:Body><' + wsdl + ':getGroups><' + wsdl + ':member><' + wsdl + ':name>';
>         soapMessage += sessionStorage.obieeUser + '</' + wsdl + ':name></' + wsdl + ':member>';
>         soapMessage += '<' + wsdl + ':expandGroups>False</' + wsdl + ':expandGroups>';
>         soapMessage += '<' + wsdl + ':sessionID>' + sessionStorage.obieeSessionId + '</' + wsdl + ':sessionID>';
>         soapMessage += '</' + wsdl + ':getGroups></soapenv:Body></soapenv:Envelope>';
>         wsCall('securityService', soapMessage, function (response) {
>             var appRoles = [];
>             var results = response.Body.getGroupsResult || { account: [] };
>             if (results.account) {
>                 results.account.forEach(function (r) {
>                     if (r.accountType == '4')
>                         appRoles.push({ id: r.name, name: r.displayName });
>                 });
>             }
>             appRoles.push({ id: "BIServiceAdministrator", name: "BIServiceAdministrator" });
>             appRoles.push({ id: "BIContentAuthor", name: "BIContentAuthor" });
>             appRoles.push({ id: "BIConsumer", name: "BIConsumer" });
>             appRoles.push({ id: "AuthenticatedUser", name: "AuthenticatedUser" });
>             successFunc(appRoles);
>         }, function (err) {
>             var appRoles = [];
>             appRoles.push({ id: "BIServiceAdministrator", name: "BIServiceAdministrator" });
>             appRoles.push({ id: "BIContentAuthor", name: "BIContentAuthor" });
>             appRoles.push({ id: "BIConsumer", name: "BIConsumer" });
>             appRoles.push({ id: "AuthenticatedUser", name: "AuthenticatedUser" });
>             successFunc(appRoles);
>         });
>     }
> ```

@serdarguler 
Thank you for your reply, the user is already logged in.
